### ISSUE TITLE
Pearson Correlation Coefficient fails when updating one batch at a time

### ISSUE THREAD

**Issue by ramonemiliani93** [Lightning-AI/torchmetrics#2014]
## 🐛 Bug

The `PearsonCorrCoef` fails when using a single batch per update.

### To Reproduce
```python
import torch
from torchmetrics import PearsonCorrCoef

metric = PearsonCorrCoef()

# Works
metric(torch.tensor([3.0, -0.5, 2.0, 7.0]), torch.tensor([2.5, 0.0, 2.0, 8.0]))
print(metric.compute())  # tensor(0.9849)

metric.reset()

# Doesn't work.
metric(torch.tensor([3.0]), torch.tensor([2.5]))
metric(torch.tensor([-0.5]), torch.tensor([0.0]))
metric(torch.tensor([2.0]), torch.tensor([2.0]))
metric(torch.tensor([7.0]), torch.tensor([8.0]))
print(metric.compute())  # tensor(nan)
```

### Expected behavior

Both ways of updating the metric should work.

### Environment
Python 3.10
torchmetrics==1.03
torch==2.01
